src/processout/processout.ts

```diff
@@ -297,20 +297,48 @@ module ProcessOut {
 
                     request.onload(<ProgressEvent>e);
                 else
-                    error(request, e);
-            };
+                    error(request, e, this.getApiRequestErrorCode(request.status));
+            }.bind(this);
             request.ontimeout  = function() {}; // Prevent IE from aborting
             request.onprogress = function() {}; // ''
             request.onabort = function() {
                 // We want to retry the call: in some cases IE fails
                 // to finish requests
-                if (retry > 3) error(request, null);
+                if (retry > 3) error(request, null, "processout-js.aborted-retries-exceeded");
                 else           this.request(method, path, data, success, error, retry + 1);
             }.bind(this);
 
             request.send(JSON.stringify(data));
         }
 
+        /**
+         * Takes an HTTP Status code and returns an ApiRequestError
+         * @param {XMLHttpRequest['status']} httpStatusCode
+         * @return {ApiRequestError}
+         */
+        protected getApiRequestErrorCode(httpStatusCode: XMLHttpRequest['status']): ApiRequestError {
+            switch (httpStatusCode) {
+                case 0:
+                    return "processout-js.network-issue"
+                case 501:
+                    return "processout-js.not-implemented"
+                case 502:
+                    return "processout-js.invalid-response"
+                case 503:
+                    return "processout-js.service-unavailable"
+                case 504:
+                    return "processout-js.response-timeout"
+                case 505:
+                    return "processout-js.http-version-not-supported"
+                case 511:
+                    return "processout-js.network-auth-required"
+
+                case 500:
+                default:
```

Comment:
Currently all status codes in [200, 500) are handled here and thus I only focussed on the 500s which were the crux of the support ticket: https://github.com/processout/processout.js/blob/44a3cc0c2806d653429964372bc7f9ac35e4c875/src/processout/processout.ts#L294

I do think you're right that we may need work on handling the 400s though - I'm not convinced the current logic of calling the onload function for them makes sense. This would definitely need more investigation